### DISCUSSION TITLE
Adapter

### DISCUSSION THREAD

**Discussion by sergeantbort** [spbrogan/sensor.mopeka_pro_check#7] [Q&A] [answered]
I have tried two Bluetooth adapters with my HA instance... one was an old one I had sitting around, didn't think to much that it didn't work, because well it was several years old anyway.  But then I bought a newer Bluetooth 5, and don't seem to be able to see any BLE devices.  

anyone has any suggestions on a USB Bluetooth dongle that works with these (amazon preferred) I would appreciate it.    I am only using it for the tank check's, as I have ESP32 for my other BLE sensors because it's better coverage.   Or possibly a way to convert over to an ESP32 for handling.

**Comment by spbrogan** [comment, marked as answer]
The bluetooth support is done with this project.  
https://github.com/TheCellule/python-bleson

But I am not convinced it is the best choice.  I have used raspberrypi 3b and 4 and both worked fine but i have seen what i think is contention when using two components that do BLE.  
If you want to keep trying i would try removing HA from the equation.  
1. Use the python interface directly in my other project and test with the example. 
https://github.com/spbrogan/mopeka_pro_check
https://github.com/spbrogan/mopeka_pro_check/blob/main/example/test_service.py
https://github.com/spbrogan/mopeka_pro_check/blob/main/docs/getting_started.md

2. Try interacting using bleson and python.  You should be able to find that in their project page. 

Also note that i have authored an esphome component for this.  I actually like using an ESP32 to get the values and make them available to HA using esphome api.  Super easy and rock solid from what i have seen.  
https://github.com/spbrogan/esphome

I am still trying to get this integrated into the main release of Esphome but it can be added easily by using the external component.  

```yml
external_components:
  - source:
      type: git
      url: https://github.com/spbrogan/esphome
      ref: mopeka_dev
    components: [ mopeka_pro_check ]
```

see PR here:
https://github.com/esphome/esphome/pull/2618

feature request:
https://github.com/esphome/feature-requests/issues/1445

Docs:
https://github.com/esphome/esphome-docs/pull/1564

**Reply by sergeantbort** [reply]
Thank you for such a quick response!  Appreciate it.    Yeah I feel the ESPhome is the way, to allow expandability currently have 3 spread around to get temp/humidity sensors so I will need to dig into getting that configured!  3 cost about the same as 1 Bluetooth dongle, so better coverage for sure. 

Looks far better than my hacked together attempt at pulling data in using the BLE client, and UUID lol.   Didn't come across it when I attempted to search for it otherwise I wouldn't have posted, asking.  Couldn't figure out how to get the python / type into the ESPhome like that!

**Comment by sergeantbort** [comment]
@spbrogan you ROCK!   I should have dug more into what was out there than trying to DIY it, lol.  been fiddling with UUID all day... with no real success...    been fiddling off and on for almost a month trying to get Bluetooth working in general (bad adapter and no other devices to really test)   But got the ESP32 boards and some of the xioami temp/humidity sensors in this week and started trying to figure it out further.   
![image](https://user-images.githubusercontent.com/82248518/147166594-29a80580-5191-479a-b1eb-813708f90a48.png)


now... it's a long shot.. but do you have any idea's or lead on the Hughes's Watchdog Surge Protector device (https://hughesautoformers.com/power-watchdog-smart-surge-protectors/)?  lol.      The manufacture doesn't have any information on the app they have for it, they contracted a company to build the app and manage it, and they have no documentation on it, been working with their General Manager who handles the product and he is hoping to get something from it so they have something in house on it vs all contracted out.     Any tips / suggestions on digging into something like that?   O

**Comment by spbrogan** [comment]
I have one of those for my RV too.  I have started trying to get the bluetooth client in esphome to connect but haven't had any luck yet.  My trailer and surge protector isn't at my house during the winter so my attempts have been limited.  But I know it isn't done using advertisements so it is going to be more work.  It requires connection.  Anyway, i am working to get a reliable vpn setup so i can connect to my trailer and update my esphome nodes so that i can prototype/play with it during the winter.  

hopefully will get that done today and then i can work on this over the next few months.   if you happen to get any documentation or anything from Hughes please share.   Glad to see the mopeka stuff is working.  I will make one more update to change the percentages to more closely match mopeka app and then hopefully get the project devs to merge into next release.

**Reply by sergeantbort** [reply]
there's a few of us in various groups working on a few things.  the only integration I have seen was through this project.. https://rvwhisper.com/

But he's in it for business, and has designed a product to sell.   I did try and reach out to him before to see if I could possibly even buy just the software to make life semi easier.. but that was a no go.

The GM at Hughes said he didn't have all of it figured out for it even. I have been working with this guy (https://github.com/rperciaccante/HA_OneControl) getting the one control stuff in.  He's got a lot of it working, we're still tweaking a bit of it, but he's similar that he's away from the rig for winter, but he's got his HA set up and talking with tierzero back home, lol.  We are in ours for winter in Florida, so I am hand's on working on stuff, im not the most experienced when it comes to the coding side, my experience comes from troubleshooting/bug testing.    We he's got the Hughes too and we are possibly going to work with the GM at some point and try and figure out what info there is and how we can move forward.

**Comment by sergeantbort** [comment]
Finally got everything connected, installed and running.  Very happy with it, appreciate your work! 
![Screenshot_20211225-112701_Home Assistant](https://user-images.githubusercontent.com/82248518/147389351-91709598-d2ec-4547-b84d-d6dbbb81bc3e.jpg)

![Screenshot_20211225-112411_Tank ](https://user-images.githubusercontent.com/82248518/147389324-3736ecbf-6b8f-482c-ac52-9963d964320f.jpg)
